### DISCUSSION TITLE
Problems trying to contrib

### DISCUSSION THREAD

**Discussion by TheTalesman** [hasura/graphql-engine#6729] [General] [answered]
Hi folks, I recently got to know about hasura at my workplace and am interested in contributing. Got some troubles however trying to setup console. My graphql engine was deployed via 1-click-deploy on heroku, my local repo has .env like this:

```
NODE_ENV=development
PORT=3000
CDN_ASSETS=true
ASSETS_PATH=https://graphql-engine-cdn.hasura.io/console/assets
ASSETS_VERSION=channel/stable/v1.0
ENABLE_TELEMETRY=true
URL_PREFIX=/
DATA_API_URL=https://---.herokuapp.com/ #my engine on heroku
#DATA_API_URL=https://localhost:8080
SERVER_VERSION=v1.0.0
CONSOLE_MODE=server
IS_ADMIN_SECRET_SET=true
```

My fork is in sync with master

When I first load the console, it opens up the field to type admin-secret. In the browser's network tab, there's a request to metadata API with the query `get_inconsistent_metadata` which gets an 404.
![Captura de tela de 2021-03-30 11-27-50](https://user-images.githubusercontent.com/4965156/113005615-03f69b00-914b-11eb-9894-5b62cf4f679a.png)

I believe this is something about metadata endpoint being /v1/metadata or /v1/query in [Endpoints.ts](https://github.com/thetalesman/graphql-engine/blob/master/console/src/Endpoints.ts#L14-L15)

Then I type my admin secret and hit "enter":
![Captura de tela de 2021-03-30 11-30-30](https://user-images.githubusercontent.com/4965156/113006005-5afc7000-914b-11eb-862d-c28ad4a46e1d.png)
The same request fails again.

Before I sync`ed my repo, I was getting errors trying to connect to datasource, which I believe comes from the same root cause causing this issue. 

If someone with more knowledge of the codebase could lend a hand would be awesome. 

Ty @tylerhannan for pointing me here from discord.

I also accept suggestions on how I could setup a development environment.

**Comment by soorajshankar** [comment, marked as answer]
Hi @TheTalesman, 
This could be because the Hasura server that you are running on heroku is older and the console codebase that you are trying to run is from the `main` branch (there are a bunch of api changes happened with Hasura 2). I think the heroku one click deployment is trying to deploy the latest stable release (1.3.3). 
If you want to run hasura console, you will need a Hasura 2 server, you can either spin up a Hasura Project on Hasura cloud or you can try deploying a newer version to heroku using the following link.

https://heroku.com/deploy?template=https://github.com/soorajshankar/graphql-engine-heroku

**Reply by soorajshankar** [reply]
technically, https://github.com/hasura/graphql-engine-heroku is the template repository that gives instructions to heroku on how to deploy Hasura, so if you want, you may fork this and try changing the version numbers in the dockerfile

**Reply by TheTalesman** [reply]
Tried new deploy with heroku and worked fine. one click deploy maybe should be updated on [docs](https://hasura.io/docs/latest/graphql/core/deployment/deployment-guides/heroku.html#heroku-one-click) 

That's great, really appreciate your help @soorajshankar.
